core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -704,13 +704,11 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
       writer.endFunCall(piFrame);
       break;
     case "TRUNC":
-      String truncFrame = getTruncFunctionName(call);
-      switch (truncFrame) {
-      case "DATE_TRUNC":
-        SqlFloorFunction.unparseDatetimeFunction(writer, call, truncFrame, false);
-        break;
-      default:
-        super.unparseCall(writer, call, leftPrec, rightPrec);
+      String truncFunctionName = getTruncFunctionName(call);
+      if (truncFunctionName.equalsIgnoreCase("DATE_TRUNC")) {
```

Comment:
Can we use a switch case instead of the if-else block?

Reply:
done